webapp/src/js/services/live-list.js

```diff
@@ -431,56 +434,45 @@ angular.module('inboxServices').factory('LiveList',
       }
       if (removeIndex !== null) {
         idx.list.splice(removeIndex, 1);
-        var removed = idx.dom.splice(removeIndex, 1);
+        const removed = idx.dom[removedItem._id];
+        delete idx.dom[removedItem._id];
 
         $(idx.selector).children().eq(removeIndex).remove();
-        if (removed.length) {
-          return removed[0];
-        }
+        return removed;
       }
     }
 
     function _setSelected(listName, _id) {
-      var i, len, doc,
-          idx = indexes[listName],
-          list = idx.list,
-          previous = idx.selected;
+      const idx = indexes[listName],
+            list = idx.list,
+            previous = idx.selected;
 
       idx.selected = _id;
 
       if (!list) {
         return;
       }
 
-      for (i=0, len=list.length; i<len; ++i) {
-        doc = list[i];
-        if (doc._id === previous) {
-          idx.dom[i]
-              .removeClass('selected');
-        }
-        if (doc._id === _id) {
-          idx.dom[i]
-              .addClass('selected');
-        }
+      if (idx.dom[_id]) {
+        idx.dom[_id].addClass('selected');
+      }
+      
+      if (previous && idx.dom[previous]) {
+        idx.dom[previous].removeClass('selected');
       }
     }
 
     function _clearSelected(listName) {
-      var i, len,
-          idx = indexes[listName],
-          list = idx.list,
-          previous = idx.selected;
+      const idx = indexes[listName];
 
-      if (!list || !previous) {
+      if (!idx.list || !idx.selected) {
         return;
       }
 
-      for (i=0, len=list.length; i<len; ++i) {
-        if (list[i]._id === previous) {
-          idx.dom[i].removeClass('selected');
-        }
+      if (idx.dom[idx.selected]) {
+        idx.dom[idx.selected].removeClass('selected');
+        delete idx.selected;
```

Comment:
This `delete` should be outside the block like previously?

Reply:
I suspect keeping it outside is safer.